staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -647,6 +647,10 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 
 	structuralSchemas := map[string]*structuralschema.Structural{}
 	for _, v := range crd.Spec.Versions {
+		if !v.Served {
```

Comment:
I think this is the problem... we need to fully construct `structuralSchemas` for all versions even if they're not being served so we have schema info for defaulting reading from storage